Makefile

```diff
@@ -1,67 +1,102 @@
-ALM_PKG := github.com/operator-framework/operator-lifecycle-manager/cmd/alm
-ALM_EXECUTABLE := ./bin/alm
-CATALOG_PKG := github.com/operator-framework/operator-lifecycle-manager/cmd/catalog
-CATALOG_EXECUTABLE := ./bin/catalog
+##########################
+#  OLM - Build and Test  #
+##########################
+
+SHELL := /bin/bash
+PKG   := github.com/operator-framework/operator-lifecycle-manager
+CMDS  := $(addprefix bin/, $(shell go list ./cmd/... | xargs -I{} basename {}))
 IMAGE_REPO := quay.io/coreos/olm
 IMAGE_TAG ?= "dev"
-PKG_DIR := pkg
 
-.PHONY: build test test-docs run clean vendor vendor-update coverage e2e
+.FORCE:
+
+.PHONY: build test run clean vendor schema-check \
+	vendor-update coverage coverage-html e2e .FORCE
 
 all: test build
 
-test-docs:
-	go test -v ./Documentation/...
+test: schema-check cover.out
 
-test-catalog:
-	go test -v ./deploy/...
+schema-check:
+	go test -v ./test/schema
 
-test: test-docs test-catalog
-	go test -v -race -coverprofile=cover.out -covermode=atomic -coverpkg ./pkg/controller/... ./pkg/...
+cover.out: schema-check
+	go test -v -race -coverprofile=cover.out -covermode=atomic \
+		-coverpkg ./pkg/controller/... ./pkg/...
 
-coverage: test
+coverage: cover.out
 	go tool cover -func=cover.out
 
-coverage-html: test
+coverage-html: cover.out
 	go tool cover -html=cover.out
 
-run-local: update-catalog
-	. ./scripts/package-release.sh ver=1.0.0-local Documentation/install/resources Documentation/install/local-values.yaml
+build: $(CMDS)
+
+$(CMDS): .FORCE
```

Comment:
You can delete the builds to trigger a rebuild, you like you can add `clean` as a pre-req to any target that you want to have a full rebuild for.

Reply:
I tried `make clean build` in the dockerfiles and it was still saying `nothing to build`? not sure if I was doing something wrong

Reply:
Hu. I don't know. That works for me locally. There's always `make clean && make build` in the Dockerfile. I was thinking like having `build: clean $(CMDS)`, then `make build` should rebuild.